src/post/post.service.ts

```diff
@@ -33,8 +33,23 @@ export class PostService {
     return `This action returns a #${id} post`;
   }
 
-  update(id: number, updatePostDto: UpdatePostDto) {
-    return `This action updates a #${id} post`;
+  async update(id: string, updatePostDto: UpdatePostDto): Promise<ResponsePostDTO> { 
+    const existingPost = await this.postModel.findByIdAndUpdate(id, updatePostDto, { new: true });
```

Comment:
What is the effect of updatePostDto and {new: true} in  this.postModel.findByIdAndUpdate(id, updatePostDto, { new: true }) ?

Reply:
It's so simple, updatePostDto is a object use to prepare data to update. {new: true} is mean the findByIdAndUpdate function will return object has updated.

Reply:
you can read [here](https://mongoosejs.com/docs/api/model.html#Model.findByIdAndUpdate())